### DISCUSSION TITLE
V2: add uncontrolled form elements

### DISCUSSION THREAD

**Discussion by johnsonthedev** [ariakit/ariakit#997] [Questions] [answered]
Since Remix, I am a big fan of uncontrolled forms and was undecided whether I like the form syntax e.g you explicitly need to add the field name to the useFormState names object. 

Just discovered https://github.com/esamattis/react-zorm and wanted to bring up this discussion. 

Do you think it would be possible to have another set of uncontrolled elements that just use the error handling ? Like focus of the first error.

**Comment by diegohaz** [comment, marked as answer]
> Do you think it would be possible to have another set of uncontrolled elements that just use the error handling ? Like focus of the first error.

Auto-focusing on the first invalid input on submit should work if you manually add `aria-invalid` to it. I don't think we'll automatically add the error and props without a controlled form.

One thing we can do to make integration with Remix more seamless, though, is to ignore the internal submit handler (and stop calling `event.preventDefault()`) if the form contains the `action` or `method` attributes.

**Reply by johnsonthedev** [reply]
nice that would help  🙏
